File: test/src/cpp_integration/cpp/aggregate.cpp
#include "aggregate.h"

using namespace cpp;

Aggregate::Aggregate(Point *point, Array<String *> *array, String *s, Number* n)
    : _point(point), _array{array}, _n(n), _s(s) {}

Point* Aggregate::getPoint() const { return _point; }
Array<String *>* Aggregate::getStringArray() const { return _array; }
String* Aggregate::getString() const { return _s; }
Number* Aggregate::getNumber() const { return _n; }

File: test/src/cpp_integration/aggregate.ts
import { Point, Aggregate } from "./declarations/cpp"

const point = new Point(1, 1);

const array = ["this", "is", "string", "array"];
const s = "this is string";
const n = 1;

const aggregate = new Aggregate(point, array, s, n);

console.assert(aggregate.getPoint().x() === 1, "Aggregate.getPoint.x failed");
console.assert(aggregate.getPoint().y() === 1, "Aggregate.getPoint.y failed");

console.assert(aggregate.getString() === "this is string", "Aggregate.getString failed");
console.assert(aggregate.getNumber() === 1, "Aggregate.getNumber failed");

const stringArrayFromAggregate = aggregate.getStringArray();
for (let i = 0; i < stringArrayFromAggregate.length; ++i) {
    console.assert(stringArrayFromAggregate[i] === array[i], "Aggregate.getStringArray failed");
}